File: Challenges/Coding World Cup/2023/Day 27 - PAK vs BAN - Bowler's Plan.cpp
char kthUnique(int n, string &s, int k)
{
    // Write your code here.
    if (k == 0)
        return '?';
    set<char> set;
    vector<char> arr;
    for (int i = 0; i < n; i++)
    {
        int currSize = set.size();
        set.insert(s[i]);
        if (currSize != set.size())
            arr.push_back(s[i]);

        if (k == arr.size())
            return arr[k - 1];
    }

    return '?';
}

File: Challenges/Coding World Cup/2023/Day 27 - PAK vs BAN - Bowler's Plan.java
import java.util.*;

public class Solution {
    static char kthUnique(int n, String s, int k) {
        if (k == 0)
            return '?';
        Set<Character> set = new HashSet<>();
        List<Character> arr = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            int currSize = set.size();
            set.add(s.charAt(i));
            if (currSize != set.size())
                arr.add(s.charAt(i));

            if (k == arr.size())
                return arr.get(k - 1);
        }

        return '?';
    }
}